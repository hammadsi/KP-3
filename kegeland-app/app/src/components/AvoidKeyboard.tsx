import React from 'react';
import {
  KeyboardAvoidingViewProps,
  KeyboardAvoidingView,
  Platform,
} from 'react-native';

/**
 * Component for avoiding the mobile keyboard from overlaying the onscreen content.
 * @param props the props
 * @see {@link KeyboardAvoidingViewProps}
 * @see {@link KeyboardAvoidingView}
 */
const AvoidKeyboard: React.FC<KeyboardAvoidingViewProps> = ({
  children,
  ...props
}) => {
  return (
    <KeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      keyboardVerticalOffset={Platform.OS === 'ios' ? 0 : 20}
      enabled={Platform.OS === 'ios'}
      {...props}>
      {children}
    </KeyboardAvoidingView>
  );
};

export default AvoidKeyboard;
