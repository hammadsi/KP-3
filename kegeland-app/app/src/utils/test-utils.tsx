/* istanbul ignore file */
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {render, RenderOptions} from '@testing-library/react-native';
import {Provider as ReduxProvider} from 'react-redux';

import {store} from '~state/store';
import {lightTheme} from '~constants/theme';
import {initialStore, mockStore} from '~state/ducks/__mocks__/store.mock';

type TestWrapperProps = {
  children: React.ReactNode;
  store?: typeof store;
};

/**
 * Wrapper component with redux state. Used when testing components
 * which requires redux.
 * @param param0 test wrapper props @see {@link TestWrapperProps}
 * @returns
 */
export const TestWrapper: React.FC<TestWrapperProps> = ({
  children,
  ...props
}) => {
  return (
    <ReduxProvider store={props.store || mockStore(initialStore)}>
      <PaperProvider theme={lightTheme}>
        <NavigationContainer theme={lightTheme}>{children}</NavigationContainer>
      </PaperProvider>
    </ReduxProvider>
  );
};

/**
 * Custom test renderer which wraps components with a redux store provider
 * @param ui the component to render
 * @param options renderer options
 * @returns test rendered componnet
 */
export const customRender = <T extends any>(
  ui: React.ReactElement<T>,
  options?: {
    wrapperProps?: {
      store: any;
    };
    renderOptions?: RenderOptions;
  },
) =>
  render(ui, {
    wrapper: (props) => <TestWrapper {...props} {...options?.wrapperProps} />,
    ...options?.renderOptions,
  });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
