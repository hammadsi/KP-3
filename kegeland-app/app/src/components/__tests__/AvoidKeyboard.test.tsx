import React from 'react';
import {Text} from 'react-native';

import {render} from '~utils/test-utils';

import AvoidKeyboard from '../AvoidKeyboard';

describe('Test AvoidKeyboard-component', () => {
  it('should render correctly', () => {
    const component = (
      <AvoidKeyboard>
        <Text>Test</Text>
      </AvoidKeyboard>
    );
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
