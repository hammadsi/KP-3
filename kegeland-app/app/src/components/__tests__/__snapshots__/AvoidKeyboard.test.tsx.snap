// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AvoidKeyboard-component should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <Text>
      Test
    </Text>
  </View>
</View>
`;
